scripts/src/altScheduleGenerator.js

```diff
@@ -1,299 +0,0 @@
- *      In particular, we assume lunch always ends 10 minutes before the next period
- * PSAT testing day is a strange anomaly
- */
-
-import iCalFetch from "./ical.js";
-import chalk from 'chalk';
-
-// Helper function to find size of an object
-Object.size = function(obj) {
-    let size = 0, key;
-    for (key in obj) {
-        if (obj.hasOwnProperty(key)) size++;
-    }
-    return size;
-};
-
-// Constants
-const EARLIEST_AM_HOUR = 6
-const HTMLnewlineRegex = /<\/?(p|div|br).*?>|\),? *(?=[A-Z0-9])/g
-const noHTMLRegex = /<.*?>/g
-const noNbspRegex = /&nbsp;/g
-const timeGetterRegex = /\(?(1?[0-9])(?::([0-9]{2}))? *(?:am)? *(?:-|–) *(1?[0-9])(?::([0-9]{2}))? *(noon|pm)?\)?/
-const newLineRegex = /\r?\n/g
-const noNewLineBeforeTimeRegex = /\n\(/g // hack for 2019-09-06 schedule
-const schoolYearStart = new Date(2021, 7, 10)
-const schoolYearEnd = new Date(2022, 5, 3)
-
-
-const altScheduleRegex = /schedule|extended|lunch/i
-const noSchoolRegex = /holiday|no\s(students|school)|break|development/i
-
-const ErrorConsole = (date) => {
-    return `${chalk.bgRedBright.black(`Error`)} on ${chalk.red.underline(date)}`;
-}
-const WarningConsole = (date) => {
-    return `${chalk.yellow(`Warning`)} on ${chalk.underline(date)}`;
-}
-
-const warnings = [];
-const errors = [];
-
-function parseAlternate (summary, description, date) { // extra date parameter is only for warnings
-    if (altScheduleRegex.test(summary)) {
-        if (!description) return undefined
-        description =
-            '\n' +
-            description
-                .replace(noNewLineBeforeTimeRegex, '(')
-                .replace(HTMLnewlineRegex, '\n')
-                .replace(noHTMLRegex, '')
-                .replace(noNbspRegex, ' ')
-        const periods = []
-        description.split(newLineRegex).map(str => {
-            let times
-            const name = str
-                .replace(timeGetterRegex, (...matches) => {
-                    times = matches
-                    return ''
-                })
-                .trim()
-
-            if (!times) {
-                /*
-                if (periods.length > 0) {
-                    periods[periods.length - 1].original += '\n' + name
-                }
-                and is */
-                return
-            }
-
-
-            let [, sH, sM = 0, eH, eM = 0, pm] = times
-
-            sH = +sH
-            sM = +sM
-            eH = +eH
-            eM = +eM
-            if (sH < EARLIEST_AM_HOUR || pm === 'pm') sH += 12
-            if (eH < EARLIEST_AM_HOUR || pm === 'pm') eH += 12
-            const startTime = sH * 60 + sM
-            const endTime = eH * 60 + eM
-
-            const duplicatePeriod = periods.findIndex(p => p.start === startTime)
-            if (~duplicatePeriod) {
-                periods[duplicatePeriod].name += '\n' + name
-                if (endTime > periods[duplicatePeriod].end)
-                    periods[duplicatePeriod].end = endTime
```

Comment:
Sometimes, especially for days with assemblies, Gunn'll list concurrent periods with different names. Something like
```
Assembly for 9th graders (9:50-10:50)
SELF for 10th graders (9:50-10:50)
Flex for others (9:50-10:50)
```
This merges the names together so a later period identification step can figure out what to show for everyone.

There was plans to make grade-specific period identification (eg show assembly vs self vs flex depending on set grade) but I never got around to that

Reply:
I see. I think the removal of this code is fine though, because it would just be rendered as 3 periods with consecutive progress bars. `useNextPeriod` would return whichever period happened to be on top so the favicon and top progress bar might be weird but besides that I think its fine. Also, as stated on discord this code hasn't been doing anything for a long time as it checks the outdated keys `start` `end` and `name` instead of `s` `e` `n`